### PR TITLE
Fix all warnings, convert to Swift 4

### PR DESCRIPTION
**Pull request by daehn**
## What's new in this PR?

* Fix all warnings, convert to Swift 4.

### CODE REVIEW

**Comment by daehn** [comment]
@mikeger Yeah, has to be done manually now, there is an option to keep it in place, but having it on results in a warning.

**Comment by vytis** [comment]
@mikeger @daehn It seems that on Swift5 it will be an error, not a warning. Maybe it's actually better to keep it as a warning for now and try to spread the work of fixing this during further development rather than do it all in one...

File: WireUtilities.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1028,6 +1029,8 @@
 				INFOPLIST_FILE = "WireUtilities-tests-host/Info.plist";
 				PRODUCT_BUNDLE_IDENTIFIER = "com.wire.$(PRODUCT_NAME:rfc1034identifier)";
 				PRODUCT_NAME = "$(TARGET_NAME)";
+				SWIFT_SWIFT3_OBJC_INFERENCE = On;
```
Comment (daehn) [inline WireUtilities.xcodeproj/project.pbxproj]: I would actually turn this off, but we can discuss that. // cc @vytis

**Comment by daehn** [comment]
Closed in favor of #49.

**Pull request closed without merging**